core/sql/src/main/java/org/trafodion/sql/HBaseClient.java

```diff
@@ -1395,6 +1422,169 @@ private boolean estimateRowCountBody(String tblName, int partialRowSize,
       return true;
     }
 
+    // Similar to estimateRowCount, except that the implementation
+    // uses a coprocessor. This is necessary when HBase encryption is
+    // in use, because the Trafodion ID does not have the proper 
+    // authorization to the KeyStore file used by HBase.
+    public boolean estimateRowCountViaCoprocessor(String tblName, int partialRowSize,
+                                    int numCols, int retryLimitMilliSeconds, long[] rc)
+                   throws ServiceException, IOException {
+      if (logger.isDebugEnabled()) {
+        logger.debug("HBaseClient.estimateRowCountViaCoprocessor(" + tblName + ") called.");
+        logger.debug("numCols = " + numCols + ", partialRowSize = " + partialRowSize);
+      }
+
+      boolean retcode = true; 
+      rc[0] = 0;
+
+      HConnection connection = null;
+      HTableInterface table = null; 
+      connection = HConnectionManager.createConnection(config);
```

Comment:
Can you use getConnection instead. It ensures that there is only one connection per process